File: typescript/lesson-5.ts
//Lesson 5: Arrays
// import { Payment } from "./lesson-4"  // import the type, since it'll said it's duplicate anw if i recreated

type Payment = { // object should be UPPERCASE, use type (instead of const)
    referenceId: string // no need comma on this
    amount: number
    paymentMethod: string
    isSettled: boolean
}

const paymentArray: Payment[] = [
    {referenceId: "txn1", amount: 10000, paymentMethod: "DANA", isSettled: true},
    {referenceId: "txn2", amount: 20000, paymentMethod: "OVO", isSettled: false},
    {referenceId: "txn3", amount: 30000, paymentMethod: "QRIS", isSettled: true}
]

function getTotalAmount ( // this is input ONLY
    payments: Payment[]
        ) { // this is where the logic goes
        let sumCalc: number = 0
        for (const payment of payments) {
            sumCalc += payment.amount 
        } 
        return sumCalc  
    }

console.log(getTotalAmount(paymentArray))

File: golang/lesson-5.go
package main

import "fmt"

type Payment struct {
	ReferenceId   string
	Amount        int
	PaymentMethod string
	IsSettled     bool
}

var payments = []Payment{
	{ReferenceId: "txn1", Amount: 10000, PaymentMethod: "OVO", IsSettled: true},
	{ReferenceId: "txn2", Amount: 20000, PaymentMethod: "DANA", IsSettled: false},
	{ReferenceId: "txn3", Amount: 30000, PaymentMethod: "QRIS", IsSettled: true},
}

func getTotalAmount(p []Payment) int { // name it first with p before the []
	var total int            // init the value to be summed
	for _, pymt := range p { // i dont understand where did i ask them to loop to the vlaue []Payment above?
		total += pymt.Amount // add always
	}
	return total
}

func main() {
	fmt.Println(getTotalAmount(payments))
}
